project_name/utils/blocks.py

```diff
@@ -30,11 +30,7 @@ class Meta:
 
 
 class ImageBlock(blocks.StructBlock):
-    image = ImageChooserBlock()
-    image_alt_text = blocks.CharBlock(
-        required=False,
-        help_text="If left blank, the image's global alt text will be used.",
-    )
+    image = WagtailImageBlock()
     caption = blocks.CharBlock(required=False)
```

Comment:
I tried migrating the project's `ImageBlock` to use Wagtail's `ImageBlock` but it would result in this very ugly set of panels:

<img width="584" alt="the image label being duplicate four times" src="https://github.com/user-attachments/assets/250e1622-fa22-4251-954c-b40503b43c92" />

This is because the model has a StreamField called `image` that consists only of this `ImageBlock`. If we use Wagtail's `ImageBlock` in place of `ImageChooserBlock`, we'll end up with another layer of `StructBlock` that has yet another block called `image` underneath it.

Reply:
FWIW, here's how the block currently looks (with `ImageChooserBlock`):

<img width="462" alt="image" src="https://github.com/user-attachments/assets/99c81181-fb3f-472a-8405-5718990aa19c" />

So, not great either, but it looks even worse with `ImageBlock`.

Reply:
That is a pretty ugly mashup of panels. We may want to revisit the drawing board on `ImageBlock` in general for this project.

Reply:
Just wanted to share that we encountered a similar issue in updating Wagtail.org to the new ImageBlock. This won't be a blocker for the website because the alt text benefits outweigh the visual weirdness. But maybe we should set up an official issue on the Wagtail repo and try to get this fixed in core?


![Screenshot 2025-03-05 at 4 42 10 PM](https://github.com/user-attachments/assets/0941cfae-9189-4c60-be92-6c12e0858720)

Reply:
Maybe, but semantically it makes sense for it to be this way. The first "Image" is the label of the group (struct block), while the second is the label for the image chooser (hence it has the required asterisk).

Handling it in core would mean we do some sort of special-casing with a logic like "If the label of the first block in a nested struct block is the same as the label of the struct block itself, hide the struct block's label", which might be a bit too much magic.